tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -163,39 +211,119 @@ void test_boundary_correction_impl(
       tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
       make_not_null(&gen), make_not_null(&pos_neg_dist), used_for_size);
-  const Scalar<DataVector> interior_normal_magnitude =
-      magnitude(interior_unit_normal_covector);
-  for (auto& t : interior_unit_normal_covector) {
-    t /= get(interior_normal_magnitude);
+  tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>
+      interior_unit_normal_vector{};
+  if constexpr (not curved_background) {
+    const Scalar<DataVector> interior_normal_magnitude =
+        magnitude(interior_unit_normal_covector);
+    for (auto& t : interior_unit_normal_covector) {
+      t /= get(interior_normal_magnitude);
+    }
+  } else {
+    using inv_spatial_metric = typename detail::inverse_spatial_metric_tag<
+        curved_background>::template f<System>;
+    // Make sure the inverse spatial metric is reasonable.
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      for (size_t j = i; j < FaceDim + 1; ++j) {
+        // NOLINTNEXTLINE
+        const_cast<DataVector&>(
+            get<inv_spatial_metric>(interior_fields_on_face).get(i, j)) *= 0.01;
+      }
+    }
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      // NOLINTNEXTLINE
+      const_cast<DataVector&>(
+          get<inv_spatial_metric>(interior_fields_on_face).get(i, i)) += 1.0;
+    }
+
+    // make the exterior inverse spatial metric be close to the interior one
+    // since the solution should be smooth.
+    std::uniform_real_distribution<> inv_metric_change_dist(0.99999, 1.0);
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      for (size_t j = i; j < FaceDim + 1; ++j) {
+        // NOLINTNEXTLINE
+        const_cast<DataVector&>(
+            get<inv_spatial_metric>(exterior_fields_on_face).get(i, j)) =
+            inv_metric_change_dist(gen) *
+            get<inv_spatial_metric>(interior_fields_on_face).get(i, j);
+      }
+    }
+
+    for (size_t i = 0; i < FaceDim + 1; ++i) {
+      interior_unit_normal_vector.get(i) =
+          get<inv_spatial_metric>(interior_fields_on_face).get(i, 0) *
+          interior_unit_normal_covector.get(0);
+      for (size_t j = 1; j < FaceDim + 1; ++j) {
+        interior_unit_normal_vector.get(i) +=
+            get<inv_spatial_metric>(interior_fields_on_face).get(i, j) *
+            interior_unit_normal_covector.get(j);
+      }
+    }
+
+    const DataVector interior_normal_magnitude = sqrt(get(dot_product(
+        interior_unit_normal_covector, interior_unit_normal_vector)));
+    for (auto& t : interior_unit_normal_covector) {
+      t /= interior_normal_magnitude;
+    }
+    for (auto& t : interior_unit_normal_vector) {
+      t /= interior_normal_magnitude;
+    }
   }
   auto exterior_unit_normal_covector = interior_unit_normal_covector;
   for (auto& t : exterior_unit_normal_covector) {
     t *= -1.0;
   }
+  auto exterior_unit_normal_vector = interior_unit_normal_vector;
+  if constexpr (curved_background) {
```

Comment:
For my understanding: why do you only reverse the sign of this in a curved background? Is it because it is only used in that context? If so, a comment might be helpful. Alternatively, always reverse it so it's always correct even if sometimes unused...

Reply:
It's unused in the Newtonian case, but I agree it's confusing. I'll just reverse it always :P Less code